Palindrome_permutation.py

```diff
@@ -1,25 +1,34 @@
-def is_permutation_of_palindrome(x):
-    # remove spaces and convert strings to lowercase
-    x = x.replace(" ", "").lower()
-
-    char_count = {}
-
-    for i in x:
-        if i.isalpha():
-            if i in char_count:
-                char_count[i] += 1
-            else:
-                char_count[i] = 1
-
-    odd_count = 0
-
-    for j in char_count.values():
-        if j % 2 != 0:
-            odd_count += 1
-
-    return odd_count <= 1
-
-a = str(input())
-print(is_permutation_of_palindrome(a))
+# In this problem we check if it's possible to obtain a mirrored string by simply rearranging the letters. "
+# If so, return - True, if not, return - False
+class Solution:
+    def __init__(self, str1):

```

Comment:
Why `str1 `and not `str` only? And maybe `input `makes more sense since it is input to the class. Also, this should be part the function. Usually in the constructor you would take input something that doesn't change within the whole scope of the class. I would encourage to read about SOLID principles from book clean code or just search it on google and watch some videos. Constructors are used for inversion of dependency. You can try to implement it in python for practice as well in future problems which would be complex involving multiple classes.